awx/main/tasks/receptor.py

```diff
@@ -510,6 +510,9 @@ def _run_internal(self, receptor_ctl):
                     raise SignalExit()
                 res = processor_future.result()
             except (SignalExit, DispatcherCancel):
+                if not signal_callback():
+                    # Record if this was from dispatcherd signal handling
+                    signal_state.dispatcher_cancel_flag = True
```

Comment:
Would prefer `dispatcher_cancel_flag` be set through some function in `SignalState`